### PR TITLE
Adding file management and admin avatar

### PR DESCRIPTION
**Pull request by loic425**

### CODE REVIEW

File: src/Entity/Media/Image.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+declare(strict_types=1);
+
+namespace App\Entity\Media;
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\MappedSuperclass
+ */
+abstract class Image extends File
```
Comment (lchrusciel) [inline src/Entity/Media/Image.php]: Are you sure this additional abstraction layer is needed?

File: src/Entity/User/AdminUser.php [hunk 2]
```diff
@@ -21,22 +21,26 @@
 class AdminUser extends BaseUser implements AdminUserInterface
 {
     /**
-     * @var string
+     * @var string|null
      *
      * @ORM\Column(type="string", nullable=true)
      */
-    protected $lastName;
+    private $lastName;
```
Comment (lchrusciel) [inline src/Entity/User/AdminUser.php]: I think you should keep these fields open

File: src/Form/Type/User/AdminAvatarType.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+namespace App\Form\Type\User;
+
+use App\Entity\User\AdminAvatar;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\FileType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class AdminAvatarType extends AbstractType
```
Comment (lchrusciel) [inline src/Form/Type/User/AdminAvatarType.php]:
```suggestion
final class AdminAvatarType extends AbstractType
```

File: src/Form/Type/User/AdminAvatarType.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
```
Comment (lchrusciel) [inline src/Form/Type/User/AdminAvatarType.php]:
```suggestion
    public function buildForm(FormBuilderInterface $builder, array $options): void
```

File: src/Form/Type/User/AdminAvatarType.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+            ->setDefaults([
+                'data_class' => AdminAvatar::class,
+            ])
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getBlockPrefix()
```
Comment (lchrusciel) [inline src/Form/Type/User/AdminAvatarType.php]:
```suggestion
    public function getBlockPrefix(): string
```

File: src/Entity/User/AdminUser.php [hunk 6]
```diff
@@ -21,22 +21,26 @@
...
      * @ORM\Column(type="string", nullable=true)
      */
-    protected $firstName;
+    private $firstName;
 
     /**
-     * AdminUser constructor.
+     * @var AdminAvatar|null
+     *
+     * @ORM\OneToOne(targetEntity="App\Entity\User\AdminAvatar", cascade={"persist"})
```
Comment (lchrusciel) [inline src/Entity/User/AdminUser.php]: Shouldn't we introduce an interface for AdminAvatar as well? In end projects, I'm ok with entities without an interface, but it is more about being coherent with rest of the classes.

File: tests/Behat/Context/Transform/AdminUserContext.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+    public function __construct(SharedStorageInterface $sharedStorage)
+    {
+        $this->sharedStorage = $sharedStorage;
+    }
+
+    /**
+     * @Transform /^(I|my)$/
+     */
+    public function getLoggedAdminUser()
```
Comment (lchrusciel) [inline tests/Behat/Context/Transform/AdminUserContext.php]:
```suggestion
    public function getLoggedAdminUser(): ?AdminUserInterface
```

File: tests/Behat/Element/Backend/TopBarElement.php [hunk 8]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
```
Comment (lchrusciel) [inline tests/Behat/Element/Backend/TopBarElement.php]: The wrong header I suppose :P

File: tests/Behat/Element/Backend/TopBarElement.php [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace App\Tests\Behat\Element\Backend;
+
+use FriendsOfBehat\PageObjectExtension\Element\Element;
+
+final class TopBarElement extends Element
```
Comment (lchrusciel) [inline tests/Behat/Element/Backend/TopBarElement.php]: I would have to check, but I suppose elements should be without the final keyword and/or should have an interface. 

```suggestion
class TopBarElement extends Element implements TopBarElementInterface
```

File: tests/Behat/Element/Backend/TopBarElement.php (hunk 8 above)
Comment (loic425) [reply tests/Behat/Element/Backend/TopBarElement.php]: lol you're right :)

File: src/Entity/User/AdminUser.php (hunk 6 above)
Comment (loic425) [reply src/Entity/User/AdminUser.php]: I added an interface for extended entities (from Sylius). But I think you're right, I could use an interface for all entities, it's indeed more coherent.

File: src/Entity/Media/Image.php (hunk 1 above)
Comment (loic425) [reply src/Entity/Media/Image.php]: You're right, I'll just keep the more generic "File" entity.

File: src/Entity/User/AdminUser.php (hunk 2 above)
Comment (loic425) [reply src/Entity/User/AdminUser.php]: Why? Admin user doesn't need to be override on the starter.

**Pull request merged**